src/charm.py

```diff
@@ -187,6 +202,12 @@ def __init__(self, *args):
             experimental_forward_auth_enabled=self._is_forward_auth_enabled,
             traefik_route_static_configs=self._traefik_route_static_configs(),
             basic_auth_user=self._basic_auth_user,
+            topology=self._topology,
+            tracing_endpoint=(
+                self._workload_tracing.get_endpoint("jaeger_thrift_http")
+                if self._is_workload_tracing_enabled()
```

Comment:
I believe `get_endpoint` internally calls `is_ready`. So, passing ` self._workload_tracing.get_endpoint("jaeger_thrift_http")` should be enough

Reply:
`_charm_tracing_config` does call `is_ready`, but `get_endpoint` doesn't, part of its docstring:

```
        It could happen that this function gets called before the provider publishes the endpoints.
        In such a scenario, if a non-leader unit calls this function, a permission denied exception will be raised due to
        restricted access. To prevent this, this function needs to be guarded by the `is_ready` check.
```

This might be something to fix in the charm library eventually, but for now I'm keeping the check.

Reply:
https://github.com/canonical/tempo-coordinator-k8s-operator/issues/72